### PR TITLE
Add additional models to provider JSON

### PR DESCRIPTION
**Pull request by robbibt**
@tsutterley I've added the following new models to the relevant provider JSON files:

- FES2012
- FES2014_extrapolated
- FES2022_extrapolated
- GOT5.6
- GOT5.6_extrapolated
- TPXO8-atlas-nc

Have tested these locally and they seem to work:
image

However, I did find that my model definitions that use wildcard matching (e.g. `"model_file": "TPXO8_atlas_v1/hf.*_tpxo8_atlas_30c_v1.nc"`) didn't recognise files in this new implementation, so I had to manually create the model file and constituent lists. I'm not experienced with this stuff so it would be great if you could double-check my changes and make sure I have specified them correctly/matched the right files to the right constituents!

### CODE REVIEW

File: providers/AVISO.json [hunk 1]
```diff
@@ -123,6 +123,83 @@
         }
     },
     "elevation": {
+        "FES2012": {
+            "format": "FES-netcdf",
+            "constituents": [
```
Comment (robbibt) [inline providers/AVISO.json]: Is specifying `constituents` necessary, or will pyTMD auto-guess the constituent names? (previously, I had simply used ` "model_file": "fes2012/data/*_FES2012_SLEV.nc"`, but that wasn't being recognised here)

File: providers/AVISO.json [hunk 2]
```diff
@@ -123,6 +123,83 @@
...
+                "mf",
+                "m4",
+                "p1",  
+                "k2",
+                "mks2",
+                "s2",
+                "m8",   
+                "nu2"
+            ],
+            "model_file": [
```
Comment (robbibt) [inline providers/AVISO.json:92]: The one file I couldn't find a match for was `Z0_FES2012_SLEV.nc` - have left it out here for now

File: providers/AVISO.json [hunk 3]
```diff
@@ -123,6 +123,83 @@
...
+                "fes2012/data/N4_FES2012_SLEV.nc",
+                "fes2012/data/O1_FES2012_SLEV.nc",
+                "fes2012/data/M3_FES2012_SLEV.nc",
+                "fes2012/data/N2_FES2012_SLEV.nc",
+                "fes2012/data/K1_FES2012_SLEV.nc",
+                "fes2012/data/S1_FES2012_SLEV.nc",
+                "fes2012/data/MS4_FES2012_SLEV.nc",
+                "fes2012/data/R2_FES2012_SLEV.nc",
+                "fes2012/data/Mu2_FES2012_SLEV.nc",
+                "fes2012/data/E2_FES2012_SLEV.nc",
```
Comment (robbibt) [inline providers/AVISO.json:109]: Is `E2` a match for `eps2`?

File: providers/AVISO.json [hunk 4]
```diff
@@ -123,6 +123,83 @@
...
+                "fes2012/data/S4_FES2012_SLEV.nc",
+                "fes2012/data/Ssa_FES2012_SLEV.nc",
+                "fes2012/data/La2_FES2012_SLEV.nc",
```
Comment (robbibt) [inline providers/AVISO.json:112]: Is `La2` a match for `l2`? Or `lambda2`?

File: providers/TPXO.json [hunk 5]
```diff
@@ -598,6 +598,27 @@
             "variable": "tide_ocean",
             "version": "8"
         },
+        "TPXO8-atlas-nc": {
+            "format": "ATLAS-netcdf",
+            "name": "TPXO8-atlas-nc",
+            "model_file": [
```
Comment (robbibt) [inline providers/TPXO.json:604]: Had previous used `"model_file": "TPXO8_atlas_v1/hf.*_tpxo8_atlas_30c_v1.nc"`, so have attempted to adapt this to a list of model files

**Comment by robbibt** [comment]
Code used to test:
``` 
from pyTMD.compute import tide_elevations
import pandas as pd
import numpy as np

fes2014 = tide_elevations(
    x=[122.269294],
    y=[-18.057430],
    delta_time=pd.date_range("2020", "2021", periods=50),
    DIRECTORY="/gdata1/data/tide_models_clipped/",
    MODEL="FES2014",
    EPSG=4326,
    TIME="datetime",
    EXTRAPOLATE=True,
    CUTOFF=np.inf,
)


# In[4]:


fes2014_extrap = tide_elevations(
    x=[122.269294],
    y=[-18.057430],
    delta_time=pd.date_range("2020", "2021", periods=50),
    DIRECTORY="/gdata1/data/tide_models_clipped/",
    MODEL="FES2014_extrapolated",
    EPSG=4326,
    TIME="datetime",
    EXTRAPOLATE=True,
    CUTOFF=np.inf,
)

fes2012 = tide_elevations(
    x=[122.269294],
    y=[-18.057430],
    delta_time=pd.date_range("2020", "2021", periods=50),
    DIRECTORY="/gdata1/data/tide_models_clipped/",
    MODEL="FES2012",
    EPSG=4326,
    TIME="datetime",
    EXTRAPOLATE=True,
    CUTOFF=np.inf,
)

fes2022_extrap = tide_elevations(
    x=[122.269294],
    y=[-18.057430],
    delta_time=pd.date_range("2020", "2021", periods=50),
    DIRECTORY="/gdata1/data/tide_models_clipped/",
    MODEL="FES2022_extrapolated",
    EPSG=4326,
    TIME="datetime",
    EXTRAPOLATE=True,
    CUTOFF=np.inf,
)

got56 = tide_elevations(
    x=[122.269294],
    y=[-18.057430],
    delta_time=pd.date_range("2020", "2021", periods=50),
    DIRECTORY="/gdata1/data/tide_models_clipped/",
    MODEL="GOT5.6",
    EPSG=4326,
    TIME="datetime",
    EXTRAPOLATE=True,
    CUTOFF=np.inf,
)

got56_extrap = tide_elevations(
    x=[122.269294],
    y=[-18.057430],
    delta_time=pd.date_range("2020", "2021", periods=50),
    DIRECTORY="/gdata1/data/tide_models_clipped/",
    MODEL="GOT5.6_extrapolated",
    EPSG=4326,
    TIME="datetime",
    EXTRAPOLATE=True,
    CUTOFF=np.inf,
)

tpxo8 = tide_elevations(
    x=[122.269294],
    y=[-18.057430],
    delta_time=pd.date_range("2020", "2021", periods=50),
    DIRECTORY="/gdata1/data/tide_models_clipped/",
    MODEL="TPXO8-atlas-nc",
    EPSG=4326,
    TIME="datetime",
    EXTRAPOLATE=True,
    CUTOFF=np.inf,
)

# Plot
pd.Series(fes2014_extrap, name="FES2014_extrapolated").plot(legend=True, figsize=(10, 5))
pd.Series(fes2012, name="FES2012").plot(legend=True)
pd.Series(fes2022_extrap, name="FES2022_extrapolated").plot(legend=True)
pd.Series(got56, name="GOT5.6").plot(legend=True)
pd.Series(got56_extrap, name="GOT5.6_extrapolated").plot(legend=True)
pd.Series(tpxo8, name="TPXO8-atlas-nc").plot(legend=True)
```

File: providers/AVISO.json (hunk 1 above)
Comment (tsutterley) [reply providers/AVISO.json]: Right, the glob string capability isn't being run for the database as I figured the database will be more "maintained".

File: providers/AVISO.json (hunk 3 above)
Comment (tsutterley) [reply providers/AVISO.json:109]: I removed the constituents from the database and am running the files through the parser as before. Should make maintenance easier :)

File: providers/AVISO.json (hunk 4 above)
Comment (tsutterley) [reply providers/AVISO.json:112]: `lambda2`

**Comment by tsutterley** [comment]
thanks @robbibt! Want to add yourself to the list of contributors?

**Pull request merged**